homeassistant/components/poolsense/__init__.py

```diff
@@ -30,20 +32,19 @@ async def async_setup(hass: HomeAssistant, config: dict):
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
     """Set up PoolSense from a config entry."""
-    poolsense = PoolSense()
+
+    poolsense = PoolSense(entry.data[CONF_EMAIL], entry.data[CONF_PASSWORD])
     auth_valid = await poolsense.test_poolsense_credentials(
-        aiohttp_client.async_get_clientsession(hass),
-        entry.data[CONF_EMAIL],
-        entry.data[CONF_PASSWORD],
+        aiohttp_client.async_get_clientsession(hass)
```

Comment:
do you think we could move session within the PoolSense as well? Thats the way id recommend and using it internally like the login info.

Reply:
Fair point. I have implemented this.